config/routes.rb

```diff
@@ -1,6 +1,12 @@
 Rails.application.routes.draw do
+  
   resources :users, only: [:index,:show] do
-    resources :posts, only: [:index,:show ]
+    resources :posts, only: [:index,:new,:create,:show ]
+  end
+
+  resources :posts do
+    resources :comments, only: [:new, :create]
```

Comment:
**OPTIONAL** - Good job done so far, but to make your work easier to read and understand, it'd be better having all your routes nested in the first route above. Please fix this 🙏
```ruby
      resources :users, only: [:index, :show] do
          resources :posts, only: [:index, :new, :create, :show] do
               resources :comments only: [:new, :create]
               resources :likes only: [:create]
          end
       end
```

Reply:
it says it is not best practice to go more than level 2 deep.